Shop_Unity_base/Assets/Scripts/Shop/Model/Items/PotionFactory.cs

```diff
@@ -1,23 +1,41 @@
-﻿
-namespace Model.Items
+﻿namespace Model.Items
 {
     using System;
     using System.Collections.Generic;
 
     public class PotionFactory : IItemFactory
     {
-        public enum PotionType { Health, Mana, Strength, Defence, Invisibility }
+        public enum PotionType { HealtSmall, HealthGrand, Mana, Strength, Poison, Invisibility }
 
         private Dictionary<PotionType, Potion> TypeToPrototype = new Dictionary<PotionType, Potion>();
 
         public PotionFactory()
         {
             string namePrefix = "P";
-            TypeToPrototype.Add(PotionType.Health, new Potion(name: $"{namePrefix}HP", cost: 10, useAmount: 2));
-            TypeToPrototype.Add(PotionType.Mana, new Potion(name: $"{namePrefix}Mana", cost: 12, useAmount: 3));
-            TypeToPrototype.Add(PotionType.Strength, new Potion(name: $"{namePrefix}Str", cost: 25, useAmount: 1));
-            TypeToPrototype.Add(PotionType.Defence, new Potion(name: $"{namePrefix}Def", cost: 25, useAmount: 1));
-            TypeToPrototype.Add(PotionType.Invisibility, new Potion(name: $"{namePrefix}Invis", cost: 30, useAmount: 1));
+            AddOrUpdatePrototype(PotionType.HealtSmall, new Potion(name: $"{namePrefix}-sHP", cost: 10),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.HealthModifier, potency: 10));
+
+            AddOrUpdatePrototype(PotionType.HealthGrand, new Potion(name: $"{namePrefix}-gHP", cost: 20),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.HealthModifier, potency: 10, useCount: 2));
+
+            AddOrUpdatePrototype(PotionType.Mana, new Potion(name: $"{namePrefix}-Mana", cost: 12),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.ManaModifier, potency: 10));
+
+            AddOrUpdatePrototype(PotionType.Strength, new Potion(name: $"{namePrefix}-Str", cost: 25),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.StrengthModifier, potency: 2, duration: 15));
+
+            AddOrUpdatePrototype(PotionType.Poison, new Potion(name: $"{namePrefix}-Poison", cost: 25),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.HealthModifier, potency: -5, useCount: 3),
+                new ThrowableComponent(range: 20));
+
+            AddOrUpdatePrototype(PotionType.Invisibility, new Potion(name: $"{namePrefix}-Invis", cost: 30),
+                new ConsumableComponent(RPGStatsComponent.StatsEffect.Invisibility, potency: 1, duration: 30));
+        }
+
+        public void AddOrUpdatePrototype(PotionType type, Potion potion, params Component<Item>[] components)
+        {
+            potion.AddComponents(components);
```

Comment:
This relies on a dumb safeguard that's based on an arbitrary rule that component owners can only hold one unique component at a time. (two failing to even add it)

Reply:
Hmm yeah I'm really not too sure how to improve this one.